### PR TITLE
fix: Use attention_mask for sub_token_lengths calculation

### PR DESCRIPTION
**Pull request by sheldon-roberts**
## Problem
Flair currently identifies padding by counting tokens that match pad_token_id. This is problematic for models such as `Qwen3-Embedding-0.6B`, where the same token (<|endoftext|>) is used for both EOS and PAD. In these cases, the EOS token at the end of a sequence is incorrectly treated as padding, causing cls_pooling="cls" to use the second-to-last token instead of the actual last token.

## Solution
A more reliable way to determine the actual sequence length is to use the attention_mask returned by the tokenizer. The attention mask correctly distinguishes real tokens (1) from padding tokens (0), regardless of whether they share the same token ID.

### CODE REVIEW

**Review by jeffpicard — APPROVED** [review]
LGTM, nice find. I think there could still be a case where a model doesn't use an attention mask, but still suffers this problem, so maybe there's a more general solution someone will find someday.

Did you confirm via local testing that the last token embedding works as expected after this change?
